### PR TITLE
chore: Support swagger + yaml

### PR DESCRIPTION
**Pull request by marikaner**
Introduced a document converter that prepares the document for parsing, returning a json openapi v3 spec.

### CODE REVIEW

File: packages/openapi-generator/package.json [hunk 1]
```diff
@@ -57,11 +57,13 @@
     "@oclif/plugin-not-found": "^1.2.3",
     "@oclif/plugin-warn-if-update-available": "^1.7.0",
     "@openapitools/openapi-generator-cli": "^2.1.4",
-    "@sap-cloud-sdk/util": "^1.33.0",
+    "@sap-cloud-sdk/util": "^1.32.1",
```
Comment (marikaner) [inline packages/openapi-generator/package.json:60]:
```suggestion
    "@sap-cloud-sdk/util": "^1.33.0",
```

File: test-packages/test-services/openapi/swagger-yaml-service/openapi/.openapi-generator/FILES [hunk 2]
```diff
@@ -8,5 +8,4 @@ configuration.ts
 git_push.sh
 index.ts
 model/index.ts
-model/simple-test-entity.ts
```
Comment (jjtang1985) [inline test-packages/test-services/openapi/swagger-yaml-service/openapi/.openapi-generator/FILES:11]: If you forget about this update, what will happen?

File: packages/openapi-generator/src/document-converter.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ */
+export async function parseFileAsJson(
+  filePath: string
+): Promise<Record<string, any>> {
+  const fileContent = await readFile(filePath, 'utf8');
+  const extension = parse(filePath).ext;
+  if (extension === '.json') {
+    return JSON.parse(fileContent);
+  }
+  if (['.yaml', '.yml'].includes(extension)) {
```
Comment (jjtang1985) [inline packages/openapi-generator/src/document-converter.ts:41]: question: will this handle capital case like `.YAML` and the json above?

File: packages/openapi-generator/src/document-converter.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+export async function convertOpenApiSpec(
+  filePath: string
+): Promise<OpenAPIV3.Document> {
+  let file;
+  try {
+    file = await parseFileAsJson(filePath);
+  } catch (err) {
+    throw errorWithCause('Could not parse OpenApi specification.', err);
+  }
+  const openApiDocument = await convertDocToOpenApiV3(file);
```
Comment (jjtang1985) [inline packages/openapi-generator/src/document-converter.ts:24]: Maybe we catch potential errors here so we understand it fails when converting from v2.0 to v3.0?

**Review by jjtang1985 — APPROVED** [review]
LGTM

File: packages/openapi-generator/src/document-converter.ts (hunk 3 above)
Comment (marikaner) [reply packages/openapi-generator/src/document-converter.ts:41]: No, I can add it, even though this is not standard ;)

File: test-packages/test-services/openapi/swagger-yaml-service/openapi/.openapi-generator/FILES (hunk 2 above)
Comment (marikaner) [reply test-packages/test-services/openapi/swagger-yaml-service/openapi/.openapi-generator/FILES:11]: What do you mean by "forget about this update"? This file is generated, I am not touching it.

File: packages/openapi-generator/src/document-converter.ts (hunk 3 above)
Comment (marikaner) [reply packages/openapi-generator/src/document-converter.ts:36]: Done.

File: packages/openapi-generator/src/document-converter.ts (hunk 4 above)
Comment (marikaner) [reply packages/openapi-generator/src/document-converter.ts:19]: Done.

File: test-packages/test-services/openapi/swagger-yaml-service/openapi/.openapi-generator/FILES (hunk 2 above)
Comment (marikaner) [reply test-packages/test-services/openapi/swagger-yaml-service/openapi/.openapi-generator/FILES:11]: @jjtang1985 I'll already merge now, but let me know if you have concerns here, that we should address. I will then either correct it or create a ticket for this.

**Pull request merged**